core/runtime.rs

```diff
@@ -2149,14 +2149,33 @@ impl JsRealm {
 
 #[inline]
 pub fn queue_async_op(
-  scope: &v8::Isolate,
+  scope: &mut v8::HandleScope,
   op: impl Future<Output = (v8::Global<v8::Context>, PromiseId, OpId, OpResult)>
     + 'static,
 ) {
-  let state_rc = JsRuntime::state(scope);
-  let mut state = state_rc.borrow_mut();
-  state.pending_ops.push(OpCall::eager(op));
-  state.have_unpolled_ops = true;
+  match OpCall::eager(op) {
+    EagerPollResult::Ready((context, promise_id, _, mut resp)) => {
+      let args = &[
+        v8::Integer::new(scope, promise_id).into(),
+        resp.to_v8(scope).unwrap(),
+      ];
+
+      let realm = JsRealm::new(context);
+      let js_recv_cb_handle =
+        realm.state(scope).borrow().js_recv_cb.clone().unwrap();
+
+      let tc_scope = &mut v8::TryCatch::new(scope);
+      let js_recv_cb = js_recv_cb_handle.open(tc_scope);
+      let this = v8::undefined(tc_scope).into();
+      js_recv_cb.call(tc_scope, this, args);
```

Comment:
Is it possible for this to result in unbounded recursion? Say I have an async op that queues another async op that queues _ad infinitum_.

Reply:
I don't think so. This calls `promise.resolve()` before the control goes back to userland JS. It works smth along the lines of:
```js
function opresolve(promiseId, ...) {
  getPromise(promiseId).resolve(...);
}

const p = setPromise();
op.op_async(promiseId, ...); // Calls `opresolve`
return p;
```

Reply:
unrelated: potential optimization by passing resolver directly instead of promiseId.

Reply:
I would personally go with passing resolver & rejecter both, and only doing `unwrapOpResult` on the error path: Ok results are already okay as dispatched, errors are the only ones that need JS time remapping. Doing the `if (err?.$err_whatever)` is slowing down the normal OK path for mostly no good reason.